server/src/main/java/org/opensearch/index/search/QueryParserHelper.java

```diff
@@ -177,7 +177,7 @@ static Map<String, Float> resolveMappingField(
     }
 
     static void checkForTooManyFields(int numberOfFields, QueryShardContext context, @Nullable String inputPattern) {
-        Integer limit = SearchModule.INDICES_MAX_CLAUSE_COUNT_SETTING.get(context.getIndexSettings().getSettings());
+        int limit = SearchService.INDICES_MAX_CLAUSE_COUNT_SETTING.get(context.getIndexSettings().getSettings());
```

Comment:
Why not have getMaxClauseCount in SearchService as [suggested earlier as well](https://github.com/opensearch-project/OpenSearch/pull/13568#discussion_r1600795498):

```
public int getMaxClauseCount() {
  return this.maxClauseCount;
}
```

Reply:
So, I think the problem there would be to make `maxClauseCount` static. If we do make it static the clause count is not dynamically update-able. I could be wrong here

Reply:
No, you're right. Also, the QueryParserHelper class is `helper`, so makes sense for it to be static and not have the SearchService instance injected. Thanks for clarifying!